dev/src/write-batch.ts

```diff
@@ -103,6 +104,39 @@ export class WriteResult {
   }
 }
 
+/** Helper type to hold a google.rpc.Status object. */
+interface RpcStatus {
+  code: Status;
+  message?: string | null;
+}
+
+/**
+ * A BatchWriteResult wraps the write time and status returned by Firestore
+ * when making BatchWriteRequests.
+ *
+ * @private
+ */
+export class BatchWriteResult {
+  constructor(
+    private readonly _writeTime: Timestamp | null,
```

Comment:
Since this is an internal type, there is no need to do the `_` + getter trick, which really only exists to provide JSDoc anyways.